### ISSUE TITLE
NaN values in loss function during Step 2.2

### ISSUE THREAD

**Issue by KHeap25** [VITA-Group/FasterSeg#54]
Hey @chenwydj ,

during the step 2.2 with [custom labels](https://github.com/VITA-Group/FasterSeg/issues/46), the value of the loss function leads to "NaN" in the last few steps (see image below).

![image](https://user-images.githubusercontent.com/66534155/102869871-8977fa00-443c-11eb-9fb2-9bbf6aea64be.png)

Everything appers to be ok till epoch 341.
![image](https://user-images.githubusercontent.com/66534155/102871938-584cf900-443f-11eb-8c4f-a6cb5bbd4ae5.png)

From epoch 351 on, something is wrong with the mIoU. All labels except "unlabeld" reach a mIoU of 0%.
![image](https://user-images.githubusercontent.com/66534155/102872037-787cb800-443f-11eb-8c52-7b63a2291957.png)

It seems to be a numerical stability problem.
Do you agree with that?
Do you have an idea to solve this problem?

It would be great if you can give me some hints to answer this questions.

Kind regards.

**Comment by Gaussianer** [comment]
Hello together (@KHeap25 @chenwydj ),
I have exactly the same problem. I also get NaN values during the training at step 2.2 at the Loss value. 

What is the reason?  Have you been able to identify the problem @KHeap25 ?

I hope I get a solution here. We would like to publish soon.

**Comment by KHeap25** [comment]
Hey @Gaussianer
Hey @chenwydj,

I have been running  the last train step (step 2.2) agian on two identically GPU's indipendent from each other. 
The result is that one GPU has finished without the NaN error and the other one (same as [above](https://github.com/VITA-Group/FasterSeg/issues/54#issue-772799205) ) gets the NaN values again.

So it seems to be a problem with my hardware / platform. 

Furthermore, I can't say which influence labels have to the train process, that are listet in [labels.py](https://github.com/mcordts/cityscapesScripts/blob/aeb7b82531f86185ce287705be28f452ba3ddbb8/cityscapesscripts/helpers/labels.py#L62) but not included in the pictures in train and val dataset.
Maybe you can give a hint, that can be interesting for the people which are working with custom data. 

If anybody will face this error/behavior again, feel free to open a new issue and cite this comments here. Perhabs we can find the origin of the NaN error.
